src/messages/proposals.rs

```diff
@@ -325,8 +344,9 @@ impl<'a> ProposalQueue<'a> {
         for queued_proposal in queued_proposal_list {
             match queued_proposal.proposal.proposal_type() {
                 ProposalType::Add => {
-                    adds.insert(queued_proposal.proposal_id().clone());
-                    proposal_queue.add(queued_proposal);
+                    let proposal_reference = queued_proposal.proposal_reference().clone();
```

Comment:
Looks like we always clone the output of this value... Shouldn't the function do this then?